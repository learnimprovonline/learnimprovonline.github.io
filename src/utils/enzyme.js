import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'
import toJson from 'enzyme-to-json'

Enzyme.configure({ adapter: new Adapter() })
