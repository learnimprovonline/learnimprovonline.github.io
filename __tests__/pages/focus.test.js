import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Link from 'gatsby-link'
import FocusPage from '../../src/pages/focus'

jest.mock(
  '../../src/data/focus/foci.json',
  () => [
    {
      name: 'One',
      description: 'A',
    },
    {
      name: 'Two',
      description: 'B',
    },
    {
      name: 'Three',
      description: 'C',
    },
  ],
  { virtual: true }
)
const mockFoci = require('../../src/data/focus/foci.json')

const FocusPageDom = shallow(<FocusPage />)

describe('Focus Page', () => {
  describe('Page Heading', () => {
    test('display text is Activity Focus', () => {
      const pageTitle = 'Activity Focus'
      const pageHeading = FocusPageDom.find('h1')
      const pageHeadingText = pageHeading.children().text()

      expect(pageHeadingText).toEqual(pageTitle)
    })
  })

  describe('Activity Page Link', () => {
    const activitiesLink = FocusPageDom.find(Link).at(0)

    test('display text is "Activities"', () => {
      const activityLinkText = activitiesLink.children().text()

      expect(activityLinkText).toEqual('Activities')
    })

    test('points to the Activity Page', () => {
      const activityPagePath = '/activities/'
      const activityLinkDestination = activitiesLink.prop('to')

      expect(activityLinkDestination).toEqual(activityPagePath)
    })
  })

  describe('Focus List', () => {
    const focusList = FocusPageDom.find('dl')

    test('description list exists', () => {
      expect(focusList.exists()).toBe(true)
    })

    test('has a term element for each focus', () => {
      const terms = focusList.find('dt')

      mockFoci.forEach((mockFocus, index) => {
        const term = terms.at(index).text()

        expect(term).toEqual(mockFocus.name)
      })
    })

    test('has a definition element for each focus', () => {
      const definitions = focusList.find('dd')

      mockFoci.forEach((mockFocus, index) => {
        const definition = definitions.at(index).text()

        expect(definition).toEqual(mockFocus.description)
      })
    })
  })

  test('renders itself as expected', () => {
    const tree = toJson(FocusPageDom)

    expect(tree).toMatchSnapshot()
  })
})
