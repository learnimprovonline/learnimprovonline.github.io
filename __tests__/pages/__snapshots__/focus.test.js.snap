// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Focus Page renders itself as expected 1`] = `
<div
  className="container"
>
  <h1>
    Activity Focus
  </h1>
  <p>
    Skill or technique used to make a performance better and more fun.
  </p>
  <GatsbyLink
    to="/activities/"
  >
    Activities
  </GatsbyLink>
  <hr />
  <dl>
    <div
      key="One"
    >
      <dt>
        One
      </dt>
      <dd>
        A
      </dd>
    </div>
    <div
      key="Two"
    >
      <dt>
        Two
      </dt>
      <dd>
        B
      </dd>
    </div>
    <div
      key="Three"
    >
      <dt>
        Three
      </dt>
      <dd>
        C
      </dd>
    </div>
  </dl>
</div>
`;
